packages/plugin/vite/src/VitePlugin.ts

```diff
@@ -105,26 +114,27 @@ export default class VitePlugin extends PluginBase<VitePluginConfig> {
 
   // Main process, Preload scripts and Worker process, etc.
   build = async (watch = false): Promise<void> => {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const that = this;
     await Promise.all(
       (
         await this.configGenerator.getBuildConfig(watch)
-      ).map((userConfig) => {
+      ).map(({ config, vite: viteConfig }) => {
         return new Promise<void>((resolve, reject) => {
           vite
             .build({
               // Avoid recursive builds caused by users configuring @electron-forge/plugin-vite in Vite config file.
               configFile: false,
-              ...userConfig,
+              ...viteConfig,
               plugins: [
                 {
                   name: '@electron-forge/plugin-vite:start',
                   closeBundle() {
                     resolve();
-
-                    // TODO: implement hot-restart here
                   },
                 },
-                ...(userConfig.plugins ?? []),
+                ...(this.isProd ? [hotRestart(config, that)] : []),
```

Comment:
issue(blocking): I think this is backwards

```suggestion
                ...(this.isProd ? [] : [hotRestart(config, that)]),
```

Reply:
You are right! 🙇